Conceptpower+Spring/src/test/java/edu/asu/conceptpower/core/ConceptManagerTest.java

```diff
@@ -15,146 +13,146 @@
+        addedConcept.setCreatorId("testuser");
+        addedConcept.setDescription("description");
+        addedConcept.setPos("noun");
+        addedConcept.setWord("test");
+        addedConcept.setWordnetId("WID-1");
+
+        Mockito.when(dbManager.getEntry("id1")).thenReturn(addedConcept);
+
+        wordnetConcept1 = new ConceptEntry();
+        wordnetConcept1.setId("WID-1");
+        wordnetConcept1.setConceptList("list1");
+        wordnetConcept1.setCreatorId("testuser");
+        wordnetConcept1.setDescription("description wid 1");
+        wordnetConcept1.setPos("noun");
+        wordnetConcept1.setWord("test wid 1");
+        wordnetConcept1.setWordnetId("WID-1");
+        wordnetConcept1.setSynonymIds("WID-2");
+
+        wordnetConcept2 = new ConceptEntry();
+        wordnetConcept2.setId("WID-2");
+        wordnetConcept2.setConceptList("list1");
+        wordnetConcept2.setCreatorId("testuser");
+        wordnetConcept2.setDescription("description wid");
+        wordnetConcept2.setPos("noun");
+        wordnetConcept2.setWord("test wid");
+        wordnetConcept2.setWordnetId("WID-2");
+
+        Mockito.when(wordNetManager.getConcept("WID-1")).thenReturn(wordnetConcept1);
+        Mockito.when(wordNetManager.getConcept("WID-2")).thenReturn(wordnetConcept2);
+
+        list1 = new ConceptList();
+        list1.setConceptListName("list1");
+        list1.setDescription("test test");
+        Mockito.when(dbManager.getConceptList("list1")).thenReturn(list1);
+
+        ConceptList wordnet = new ConceptList();
+        wordnet.setConceptListName(Constants.WORDNET_DICTIONARY);
+        wordnet.setDescription("WordNet list");
+        Mockito.when(dbManager.getConceptList(Constants.WORDNET_DICTIONARY)).thenReturn(wordnet);
+    }
+
+    @Test
+    public void testGetConceptEntryForAddedConcept() {
+        ConceptEntry entry = managerToTest.getConceptEntry("id1");
+        Assert.assertEquals(addedConcept, entry);
+
+        Assert.assertTrue(entry.getSynonymIds().contains("WID-2"));
+    }
+
+    @Test
+    public void testGetConceptEntryForWordnetConcept() {
+        ConceptEntry entry = managerToTest.getConceptEntry("WID-2");
+        Assert.assertEquals(wordnetConcept2, entry);
+    }
+
+    @Test
+    public void testGetConceptEntryNoExisting() {
+        ConceptEntry entry = managerToTest.getConceptEntry("id-non");
+        Assert.assertNull(entry);
+    }
+
+    @Test
+    public void testAddConceptListEntry() throws DictionaryDoesNotExistException, DictionaryModifyException {
+        ConceptEntry newConcept = new ConceptEntry();
+        newConcept.setConceptList("list1");
+        newConcept.setCreatorId("testuser");
+        newConcept.setDescription("description");
+        newConcept.setPos("noun");
+        newConcept.setWord("test new");
+        newConcept.setWordnetId("WID-1");
+
+        String id = managerToTest.addConceptListEntry(newConcept);
+        Assert.assertNotNull(id);
```

Comment:
@jdamerow  Should i write a separate test class for dbmanager and test the methods of dbmanager. Because the dbmanager is just storing the values to the db and not returning anything. Or do you want me to mock dbmanager (store method)